### ISSUE TITLE
Remove deprecated features

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#1310]
- [x] `emptyLineBefore` option in `declaration-block-properties-order`
- [x] `hierarchicalSelectors` option in `indentation`
- [x] `cssWordIsVariable` util
- [x] namespaceless plugin rule names
- [x] contextual quote rule options

**Comment by ntwb** [comment]
Is the plan to remove support but keep the notices?

Scenario: Someone is using stylelint `6.0.0` and with the release of `7.0.0` they update and start to update there stylelint configuration for `7.0.0` compatibility. As they had never tested against stylelint `6.5.1` they never saw the deprecation notices regarding _contextual quote rule options_ that in `6.5.1`. All they will see is:

```shell
Invalid Option: Invalid option value "double-where-recommended" for rule "font-family-name-quotes"
```

Is there a way where the message would instead show something similar:
```shell
Invalid Option: Option value "double-where-recommended" for rule "font-family-name-quotes" was deprecated in stylelint 7.0.0
```

**Comment by davidtheclark** [comment]
That would not be a trivial change ... I'd rather people read the changelog. Maybe we should add a little more to the standard Invalid Option message stating that it may be a typo, or it may be a feature that has been removed and you should check the changelog.

**Comment by ntwb** [comment]
> That would not be a trivial change ... 

Right 👍 

> I'd rather people read the changelog. Maybe we should add a little more to the standard Invalid Option message stating that it may be a typo, or it may be a feature that has been removed and you should check the changelog.

Sounds good.

**Comment by davidtheclark** [comment]
Closed by #1385.
